instrumentation/undertow-1.4/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/undertow/v1_4/UndertowHttpServerExchangeInstrumentation.java

```diff
@@ -0,0 +1,80 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.instrumentation.hypertrace.undertow.v1_4;
+
+import static net.bytebuddy.matcher.ElementMatchers.failSafe;
+import static net.bytebuddy.matcher.ElementMatchers.isPublic;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.javaagent.instrumentation.api.ContextStore;
+import io.opentelemetry.javaagent.instrumentation.api.InstrumentationContext;
+import io.opentelemetry.javaagent.instrumentation.undertow.UndertowHttpServerTracer;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import io.undertow.server.HttpServerExchange;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatcher.Junction;
+import org.hypertrace.agent.core.instrumentation.SpanAndObjectPair;
+import org.xnio.channels.StreamSourceChannel;
+
+public final class UndertowHttpServerExchangeInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return failSafe(named("io.undertow.server.HttpServerExchange"));
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    final Map<Junction<MethodDescription>, String> matchers = new HashMap<>();
+    matchers.put(
+        named("getRequestChannel")
+            .and(takesArguments(0))
+            .and(returns(named("org.xnio.channels.StreamSourceChannel")))
+            .and(isPublic()),
+        UndertowHttpServerExchangeInstrumentation.class.getName() + "$getRequestChannel_advice");
+    return Collections.unmodifiableMap(matchers);
+  }
+
+  /** TODO */
+  static final class getRequestChannel_advice {
+
+    @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+    public static void exit(
+        @Advice.This final HttpServerExchange thizz,
+        @Advice.Return final StreamSourceChannel returnedChannel) {
+      final Context serverContext = UndertowHttpServerTracer.tracer().getServerContext(thizz);
+      final Span span = Span.fromContext(serverContext);
+      final SpanAndObjectPair spanAndObjectPair =
+          new SpanAndObjectPair(Objects.requireNonNull(span, "span should not be null"));
```

Comment:
`Objects.requireNonNull(span, "span should not be null")` span is never null. `Span.fromContext(serverContext)` returns either recording span or default one.